gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -304,7 +297,7 @@ def self.run_sig(hook_mod, method_name, original_method, declaration_block)
         Signature.new_untyped(method: original_method)
       end
 
-    unwrap_method(hook_mod, signature, original_method)
+    unwrap_method(signature.method.owner, signature, original_method)
```

Comment:
This ensures that in the non-alias case, we don't change behavior, since the thing we pass as `mod` will now actually be used